lib/daru/core/group_by.rb

```diff
@@ -13,7 +13,15 @@ def each_group
 
       TUPLE_SORTER = lambda do |a, b|
         if a && b
-          a.compact <=> b.compact
+          a_comp = a.compact
+          b_comp = b.compact
+          a_comp_len = a_comp.length
+          b_comp_len = b_comp.length
+          if a_comp_len == b_comp_len
+            a_comp <=> b_comp || 0
+          else
+            0
+          end
         else
           a ? 1 : -1
         end
```

Comment:
WDYT about this version?
```ruby
TUPLE_SORTER = lambda do |left, right|
  return -1 if !right
  return 1 if !left

  left = left.compact
  right = right.compact
  return left <=> right if left.lenght == right.length
  left.length <=> right.length
end
```
It looks a bit more readable.

Reply:
Done with necessary corrections